cmd/config/internal/commands/e2e/e2econtainerconfig/main.go

```diff
@@ -6,54 +6,54 @@ package main
 import (
 	"fmt"
 	"os"
+	"strconv"
 
+	// corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/kustomize/kyaml/fn/framework"
+	"sigs.k8s.io/kustomize/kyaml/fn/framework/command"
+	"sigs.k8s.io/kustomize/kyaml/kio"
 	"sigs.k8s.io/kustomize/kyaml/yaml"
 )
 
-// Data contains the items
-type Data struct {
-	StringValue string `yaml:"stringValue,omitempty"`
-
-	IntValue int `yaml:"intValue,omitempty"`
-
-	BoolValue bool `yaml:"boolValue,omitempty"`
-}
-
-// Example defines the ResourceList.functionConfig schema.
-type Example struct {
+// Config defines the ResourceList.functionConfig schema.
+type Config struct {
 	// Data contains configuration data for the Example
 	// Nest values under Data so that the function can accept a ConfigMap as its
 	// functionConfig (`run` generates a ConfigMap for the functionConfig when run with --)
 	// e.g. `config run DIR/ --image my-image -- a-string-value=foo` will create the input
 	// with ResourceList.functionConfig.data.a-string-value=foo
-	Data Data `yaml:"data,omitempty"`
+	Data map[string]string `yaml:"data,omitempty"`
 }
 
 func main() {
-	functionConfig := &Example{}
-	resourceList := &framework.ResourceList{FunctionConfig: functionConfig}
-
-	cmd := framework.Command(resourceList, func() error {
-		for i := range resourceList.Items {
-			if err := resourceList.Items[i].PipeE(yaml.SetAnnotation("a-string-value",
-				functionConfig.Data.StringValue)); err != nil {
-				return err
+	// config is same structure to configMap.
+	// config := &corev1.ConfigMap{}
+	config := &Config{}
+
+	fn := func(items []*yaml.RNode) ([]*yaml.RNode, error) {
+		var newNodes []*yaml.RNode
+		for i := range items {
+			// set the annotation on each resource item
+			if err := items[i].PipeE(yaml.SetAnnotation("a-string-value", config.Data["stringValue"])); err != nil {
+				return nil, fmt.Errorf("%w", err)
 			}
-
-			if err := resourceList.Items[i].PipeE(yaml.SetAnnotation("a-int-value",
-				fmt.Sprintf("%v", functionConfig.Data.IntValue))); err != nil {
-				return err
+			intValue, _ := strconv.Atoi(config.Data["intValue"])
```

Comment:
Please check the returned error here; in general we don't ignore any potential errors

Reply:
I forgot to write this. I fix it now.